### ISSUE TITLE
Stock Price Analysis

### ISSUE THREAD

**Issue by keshripritesh** [SrigadaAkshayKumar/stock#188]
# 📈 Stock Price Analysis

A machine learning project focused on analyzing and forecasting stock market prices using **time series forecasting techniques**. This project is part of the **GirlScript Summer of Code (GSSoC’25)** contribution.

---

## 🎯 **Aim**

The primary aim of this project is to build a forecasting model to analyze **Nifty50 stock market prices** and provide predictive insights using **ARIMA time series models**.

---

## 📂 **Dataset**

* The dataset can be downloaded from [[Nifty50 Stock Market Data (Kaggle)](https://www.kaggle.com/rohanrao/nifty50-stock-market-data)](https://www.kaggle.com/rohanrao/nifty50-stock-market-data).
* It contains historical stock prices of Nifty50 companies.

---

## 🛠️ **Steps Involved**

* Importing and exploring the dataset
* Data visualization for better understanding of stock patterns
* Data preprocessing and handling missing values
* Model training using **ARIMA**
* Model evaluation with relevant metrics

---

## 🤖 **Model Used**

* **ARIMA (Auto-Regressive Integrated Moving Average)** for time series forecasting.

---

## 📚 **Libraries Used**

* `pandas`
* `numpy`
* `os`
* `seaborn`
* `matplotlib`
* `statsmodels`
* `sklearn`

---

## ✅ **Evaluation Metrics**

```
Testing Mean Squared Error (MSE): 1431.999
Symmetric Mean Absolute Percentage Error (SMAPE): 1.503
```

---

## 📌 **Conclusion**

The ARIMA model achieved **low error values**, showing its ability to capture stock price patterns effectively. While it performs well, further improvements can be made using **LSTM, Prophet, or Hybrid models** for better accuracy.

---

## 🔮 **Future Scope**

* Implement **LSTM / GRU neural networks** for deep learning-based time series forecasting
* Add **Prophet** model for easy interpretability
* Create a **Streamlit web app** to visualize and interact with stock predictions
* Expand the dataset to include **global indices and stock data**

---

## 📸 **Output Snapshots**

<img width="730" height="441" alt="Image" src="https://github.com/user-attachments/assets/780e17c0-f659-4963-9fa4-496d35db68af" />

<img width="445" height="352" alt="Image" src="https://github.com/user-attachments/assets/61141a6f-b00d-45b1-a1f5-a60b383792e4" />

---

**Comment by keshripritesh** [comment]
Hi @SrigadaAkshayKumar 👋,

This issue (#188) provides the detailed context and description of the project "Stock Price Analysis" — including aim, dataset, steps, model used, evaluation, and future scope.

The corresponding PR (#189) contains the actual implementation of this project (code, model training, evaluation, and output snapshots). Together, the issue documents the problem statement and scope, while the PR delivers the solution. 🚀
